packages/belos/tpetra/test/MINRES/test_minres_hb.cpp

```diff
@@ -0,0 +1,302 @@
+
+    //
+    // *****Construct initial guess and random right-hand-sides *****
+    //
+    RCP<tmultivector_t> B, X;    
+    int tmp_numRHS = numRHS;
+    if (tmp_numRHS != numRHS)
+    {
+      X = rcp( new MV(rowMap, numRHS) );
+      MVT::MvRandom( *X );
+      B = rcp( new MV(rowMap, numRHS ) );
+      OPT::Apply( *A, *X, *B );
+      MVT::MvInit( *X, 0.0 );
+    }
+    
+    //
+    // Compute the initial residual norm of the problem, so we can see
+    // by how much it improved after the solve.
+    //
+    std::vector<double> initialResidualNorms (numRHS);
+    std::vector<double> initialResidualInfNorms (numRHS);
+    MV R(rowMap, numRHS);
+    OPT::Apply (*A, *X, R);
+    MVT::MvAddMv (-1.0, R, 1.0, *B, R); // R := -(A*X) + B.
+    MVT::MvNorm (R, initialResidualNorms);
+    MVT::MvNorm (R, initialResidualInfNorms, Belos::InfNorm);
+    if (verbose) {
+      verbOut << "Initial residual 2-norms:            \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << initialResidualNorms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl << "Initial residual Inf-norms:          \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << initialResidualInfNorms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl;
+    }
+    
+    std::vector<double> rhs2Norms (numRHS);
+    std::vector<double> rhsInfNorms (numRHS);
+    MVT::MvNorm (*B, rhs2Norms);
+    MVT::MvNorm (*B, rhsInfNorms, Belos::InfNorm);
+    if (verbose) {
+      verbOut << "Right-hand side 2-norms:             \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << rhs2Norms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl << "Right-hand side Inf-norms:           \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << rhsInfNorms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl;
+    }
+    
+    std::vector<double> initialGuess2Norms (numRHS);
+    std::vector<double> initialGuessInfNorms (numRHS);
+    MVT::MvNorm (*X, initialGuess2Norms);
+    MVT::MvNorm (*X, initialGuessInfNorms, Belos::InfNorm);
+    if (verbose) {
+      verbOut << "Initial guess 2-norms:               \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << initialGuess2Norms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl << "Initial guess Inf-norms:             \t";
+      for (int i = 0; i < numRHS; ++i) {
+        verbOut << initialGuessInfNorms[i];
+        if (i < numRHS-1) {
+          verbOut << ", ";
+        }
+      }
+      verbOut << std::endl;
+    }
+    
+    //
+    // Compute the infinity-norm of A.
+    //
+    // AM: No equivalence in Tpetra for normInf() ...
+    //const double normOfA = A->NormInf();
```

Comment:
There is no equivalence in Tpetra (or I didn't find it) of Epetra `NormInf()`.

What should I do?

Reply:
@antoinemeyer5 Have you tried this? https://docs.trilinos.org/dev/packages/tpetra/doc/html/classTpetra_1_1Vector.html#abfdec9f59f76bd4f7c234a5d5c152ce4

Reply:
Yes, but unfortunately this is for the `Tpetra::Vector` class. Here A is a `Tpetra::CrsMatrix`